pkg/controller/infrastructure/add.go

```diff
@@ -38,13 +38,15 @@ type AddOptions struct {
 	MachineImageOwnerSecretRef *corev1.SecretReference
 	// WhitelistedImageIDs specifies an array of image IDs that will bypass image sharing.
 	WhitelistedImageIDs []string
+	// EnableSshKeyRotation ensures ssh key rotation always succeeds by detaching all VM instances using the old ssh key.
+	EnableSshKeyRotation bool
```

Comment:
Is it needed after https://github.com/gardener/gardener-extension-provider-alicloud/pull/339#discussion_r694072155?

```suggestion
```